scripts/backref_cost_attribution_v9_transport.py

```
@@ -0,0 +1,676 @@
+    if root != home / SAFE_ROOT_BASENAME or child.parent != root or \
+            not child.name.startswith(RUN_PREFIX) or len(child.name) != 12:
+        raise FrameRefusal("run directory is not the exact v9 safe-root child")
+    info = os.lstat(child)
+    if stat.S_ISLNK(info.st_mode) or not stat.S_ISDIR(info.st_mode) or \
+            Path(os.path.realpath(child)) != child:
+        raise FrameRefusal("run directory is not canonical and real")
+
+
+def _run(argv: list[str], *, cwd: Path | None = None,
+         env: dict[str, str] | None = None) -> subprocess.CompletedProcess[bytes]:
+    return subprocess.run(argv, cwd=cwd, env=env, stdin=subprocess.DEVNULL,
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)